nexus/tests/integration_tests/instances.rs

```diff
@@ -1798,6 +1803,99 @@ async fn test_attach_one_disk_to_instance(cptestctx: &ControlPlaneTestContext) {
     assert_eq!(disks[0].state, DiskState::Attached(instance.identity.id));
 }
 
+#[nexus_test]
+async fn test_instance_fails_to_boot_with_disk(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    // Test that the saga correctly unwinds if the sled_agent's instance_put fails
+    // see: https://github.com/oxidecomputer/omicron/issues/1713
+    let client = &cptestctx.external_client;
+
+    const POOL_NAME: &str = "p0";
+    const ORGANIZATION_NAME: &str = "bobs-barrel-of-bytes";
+    const PROJECT_NAME: &str = "bit-barrel";
+
+    // Test pre-reqs
+    DiskTest::new(&cptestctx).await;
+    create_ip_pool(&client, POOL_NAME, None, None).await;
+    create_organization(&client, ORGANIZATION_NAME).await;
+    create_project(client, ORGANIZATION_NAME, PROJECT_NAME).await;
+
+    // Create the "probablydata" disk
+    create_disk(&client, ORGANIZATION_NAME, PROJECT_NAME, "probablydata").await;
+
+    // Verify disk is there and currently detached
+    let url_project_disks = format!(
+        "/organizations/{}/projects/{}/disks",
+        ORGANIZATION_NAME, PROJECT_NAME,
+    );
+    let disks: Vec<Disk> = NexusRequest::iter_collection_authn(
+        client,
+        &url_project_disks,
+        "",
+        None,
+    )
+    .await
+    .expect("failed to list disks")
+    .all_items;
+    assert_eq!(disks.len(), 1);
+    assert_eq!(disks[0].state, DiskState::Detached);
+
+    // Create the instance
+    let instance_params = params::InstanceCreate {
+        identity: IdentityMetadataCreateParams {
+            name: Name::try_from(String::from("nfs")).unwrap(),
+            description: String::from("probably serving data"),
+        },
+        ncpus: InstanceCpuCount::try_from(32).unwrap(),
+        memory: ByteCount::from_gibibytes_u32(4),
+        hostname: String::from("nfs"),
+        user_data: vec![],
+        network_interfaces: params::InstanceNetworkInterfaceAttachment::Default,
+        external_ips: vec![],
+        disks: vec![params::InstanceDiskAttachment::Attach(
+            params::InstanceDiskAttach {
+                name: Name::try_from(String::from("probablydata")).unwrap(),
+            },
+        )],
+        start: true,
+    };
+
+    let url_instances = format!(
+        "/organizations/{}/projects/{}/instances",
+        ORGANIZATION_NAME, PROJECT_NAME
+    );
+
+    let builder =
+        RequestBuilder::new(client, http::Method::POST, &url_instances)
+            .body(Some(&instance_params))
+            .expect_status(Some(http::StatusCode::INTERNAL_SERVER_ERROR));
```

Comment:
What will actually cause this instance create to fail? The setup looks the same as in `test_attach_one_disk_to_instance` above which should succeed.

Reply:
Ah, I see the `> 16` vCPUs bit :P Add a comment to indicate why this should fail here?

Reply:
added comments in be9f7a37